environment/stats/src/cloud/benchflow/collectors/stats.go

```diff
@@ -110,34 +131,16 @@ func stopCollecting(w http.ResponseWriter, r *http.Request) {
 		fmt.Fprintf(w, "Currently not collecting")
 		return
 	}
+	for _, c := range containers {
+		c.stopChannel <- true
+		}
 	collecting = false
 	fmt.Fprintf(w, "Stopped collecting")
 }
 
 func main() {
 	collecting = false
```

Comment:
I guess you can get rid of the collecting variable if you implement the WaitGroup